src/neural_network/NeuralNetwork.cpp

```diff
@@ -70,7 +70,7 @@ void NeuralNetwork::trainOnce(const vector<float>& inputs, const vector<float>&
 
 vector<float> NeuralNetwork::output(const vector<float>& inputs, bool temporalReset)
```

Comment:
Question: don't you need change the input parameters? like: 

```suggestion
vector<float> NeuralNetwork::output(const Tensor & inputs, bool temporalReset)
```

In order to avoid the cast

Reply:
I should but I don't want to expose the Tensor class to the public API.